File: Iteration2/code/CSE3063GRP12/src/core/models/concretes/Student.java
package core.models.concretes;

import java.util.ArrayList;
import core.models.abstracts.User;

public class Student extends User {
    private Advisor advisor;
    private ArrayList<CourseSession> listOfLectureSessions;
    private Transcript transcript;

    public Student(String id, String firstName, String lastName, String userName,
            String password, Advisor advisor, ArrayList<CourseSession> listOfLectureSessions,
            Transcript transcript) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.userName = userName;
        this.password = password;
        this.advisor = advisor;
        this.listOfLectureSessions = listOfLectureSessions;
        this.transcript = transcript;
    }

    public Student() {
    }

    public ArrayList<CourseSession> getListOfLectureSessions() {
        return listOfLectureSessions;
    }

    public Advisor getAdvisor() {
        return advisor;
    }

    public Transcript getTranscript() {
        return transcript;
    }

    public void setAdvisor(Advisor advisor) {
        this.advisor = advisor;
    }

    public void setListOfLectureSessions(ArrayList<CourseSession> listOfLectureSessions) {
        this.listOfLectureSessions = listOfLectureSessions;
    }

    public void setTranscript(Transcript transcript) {
        this.transcript = transcript;
    }

}

File: Iteration3/code/CSE3063GRP12/src/core/models/concretes/Student.py
from core.models.abstracts.User import User

class Student(User):
    def __init__(self, id=None, firstName=None, lastName=None, userName=None, password=None, advisor=None, listOfLectureSessions=None, transcript=None):
        super().__init__(id, firstName, lastName, userName, password)
        self.advisor = advisor
        self.listOfLectureSessions = listOfLectureSessions if listOfLectureSessions else []
        self.transcript = transcript
    
    def to_dict(self):
        return {
            "id": self.id,
            "firstName": self.firstName,
            "lastName": self.lastName,
            "userName": self.userName,
            "password": self.password,
            "advisor": self.advisor.to_dict() if self.advisor else None,
            "listOfLectureSessions": [lectureSession.to_dict() if hasattr(lectureSession, 'to_dict') else lectureSession for lectureSession in self.listOfLectureSessions],
            "transcript": [course.to_dict() if hasattr(course, 'to_dict') else course for course in self.transcript]
        }